v11y/index/src/lib.rs

```diff
@@ -0,0 +1,296 @@
+use core::str::FromStr;
+use cve::{common, published, Cve, Published, Rejected};
+use cvss::v3::Base;
+use sikula::{mir::Direction, prelude::*};
+use tantivy::{
+    collector::TopDocs,
+    query::{AllQuery, BooleanQuery, TermQuery, TermSetQuery},
+    schema::INDEXED,
+    store::ZstdCompressor,
+    DocAddress, IndexSettings, Order, Searcher, SnippetGenerator,
+};
+use time::{format_description::well_known::Rfc3339, OffsetDateTime};
+use trustification_api::search::SearchOptions;
+use trustification_index::{
+    boost, create_boolean_query, create_date_query, create_string_query, field2float, field2str, field2strvec,
+    metadata::doc2metadata,
+    tantivy::{
+        doc,
+        query::{Occur, Query},
+        schema::{Field, Schema, Term, FAST, STORED, STRING, TEXT},
+        DateTime,
+    },
+    term2query, Document, Error as SearchError,
+};
+use v11y_model::search::{Cves, CvesSortable, SearchDocument, SearchHit};
+
+pub struct Index {
+    schema: Schema,
+    fields: Fields,
+}
+
+struct Fields {
+    indexed_timestamp: Field,
+
+    id: Field,
+    state: Field,
+
+    title: Field,
+    description: Field,
+
+    cvss31_score: Field,
+    cvss30_score: Field,
+}
+
+impl Default for Index {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Index {
+    pub fn new() -> Self {
+        let mut schema = Schema::builder();
+        let fields = Fields {
+            indexed_timestamp: schema.add_date_field("indexed_timestamp", STORED),
+            id: schema.add_text_field("id", STRING | FAST | STORED),
+            state: schema.add_text_field("state", STRING | FAST | STORED),
+
+            title: schema.add_text_field("title", TEXT | FAST | STORED),
+            description: schema.add_text_field("description", TEXT | FAST | STORED),
+
+            cvss31_score: schema.add_f64_field("cvss31_score", FAST | INDEXED | STORED),
+            cvss30_score: schema.add_f64_field("cvss30_score", FAST | INDEXED | STORED),
```

Comment:
Building an index for floats is probably not needed, so just FAST | STORED would do.